internal/venice-common/src/main/java/com/linkedin/venice/pubsub/api/PubSubPosition.java

```diff
@@ -0,0 +1,41 @@
+package com.linkedin.venice.pubsub.api;
+
+import com.linkedin.venice.pubsub.PubSubPositionFactory;
+
+
+/**
+ * Represents a position of a message in a partition of a topic.
+ */
+public interface PubSubPosition {
```

Comment:
Is the intent that eventually, `PubSubMessage<K, V, OFFSET>` will become `PubSubMessage<K, V, OFFSET extends PubSubPosition>`? It's ok to do it separately in another commit, but just looking to see IIUC.

Reply:
Since we're using the interface for representing the position I believe we do not need to have `OFFSET` in `PubSubMessage<K, V, OFFSET>`. In fact, I would like to get rid of `K` and `V` as well from `PubSubMessage`.